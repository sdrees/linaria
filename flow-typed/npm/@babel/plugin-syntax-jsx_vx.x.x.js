// flow-typed signature: 277e3badb8f654ae1d6debe273f77ef5
// flow-typed version: <<STUB>>/@babel/plugin-syntax-jsx_v^7.2.0/flow_v0.93.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@babel/plugin-syntax-jsx'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@babel/plugin-syntax-jsx' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@babel/plugin-syntax-jsx/lib/index' {
  declare module.exports: any;
}

// Filename aliases
declare module '@babel/plugin-syntax-jsx/lib/index.js' {
  declare module.exports: $Exports<'@babel/plugin-syntax-jsx/lib/index'>;
}
