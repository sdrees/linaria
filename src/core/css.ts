type CSSProperties = {
  [key: string]: string | number | CSSProperties;
};

export default function css(
  _strings: TemplateStringsArray,
  ..._exprs: Array<string | number | CSSProperties>
): string {
  throw new Error(
    'Using the "css" tag in runtime is not supported. Make sure you have set up the Babel plugin correctly.'
  );
}
