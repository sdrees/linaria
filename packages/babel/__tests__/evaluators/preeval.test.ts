import { join } from 'path';
import { transformAsync } from '@babel/core';
import dedent from 'dedent';
import serializer from '../../__utils__/linaria-snapshot-serializer';
import type { StrictOptions } from '../../src';

expect.addSnapshotSerializer(serializer);

const options: Partial<StrictOptions> = {
  displayName: true,
  evaluate: true,
};

const transpile = async (input: string) =>
  (await transformAsync(input, {
    babelrc: false,
    presets: [[require.resolve('@linaria/preeval'), options]],
    plugins: [
      '@babel/plugin-proposal-class-properties',
      '@babel/plugin-syntax-jsx',
    ],
    filename: join(__dirname, 'app/index.js'),
    configFile: false,
  }))!;

it('preserves classNames', async () => {
  const { code } = await transpile(
    dedent`
      import { styled } from '@linaria/react';

      const Component = styled.div\`\`;
      `
  );

  expect(code).toMatchSnapshot();
});

it('handles locally named import', async () => {
  const { code } = await transpile(
    dedent`
      import { styled as custom } from '@linaria/react';

      const Component = custom.div\`\`;
      `
  );

  expect(code).toMatchSnapshot();
});

it('replaces functional component', async () => {
  const div = '<div>{props.children}</div>';
  const { code } = await transpile(
    dedent`
      import React from 'react';

      const Component = (props) => ${div};
      `
  );

  expect(code).toMatchSnapshot();
});

it('replaces class component', async () => {
  const div = '<div>{props.children}</div>';
  const { code } = await transpile(
    dedent`
      import React from 'react';

      class Component extends React.PureComponent {
        render() {
          return ${div};
        }
      }
      `
  );

  expect(code).toMatchSnapshot();
});

it('replaces constant', async () => {
  const div = '<div>{props.children}</div>';
  const { code } = await transpile(
    dedent`
      import React from 'react';

      const tag = ${div};

      const Component = (props) => tag;
      `
  );

  expect(code).toMatchSnapshot();
});
