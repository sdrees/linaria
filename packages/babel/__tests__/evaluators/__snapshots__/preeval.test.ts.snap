// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles locally named import 1`] = `
"import { styled as custom } from '@linaria/react';
const Component =
/*linaria ci72cjv Component Component_ci72cjv*/
custom.div\`\`;"
`;

exports[`preserves classNames 1`] = `
"import { styled } from '@linaria/react';
const Component =
/*linaria ci72cjv Component Component_ci72cjv*/
styled.div\`\`;"
`;

exports[`replaces class component 1`] = `
"import React from 'react';

function Component() {
  return <></>;
}"
`;

exports[`replaces constant 1`] = `
"import React from 'react';
const tag = <></>;

const Component = props => tag;"
`;

exports[`replaces functional component 1`] = `
"import React from 'react';

const Component = () => {
  return <></>;
};"
`;
